timelog2html

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/lua5.1
+
+args = {...}
+
+vfile = assert(args[1], "arg1 missing: .v file")
+table.remove(args,1)
+assert(#args > 0, "arg missing: at lease one aux file")
+data_files = args
+
+source = assert(io.open(vfile), "unable to open "..vfile):read("*a")
+
+function htmlescape(s)
+	return (s:gsub("&","&amp;"):gsub("<","&lt;"):gsub(">","&gt;"))
+end
+
+colors = {
+  '#F08080', '#EEE8AA', '#98FB98'
+}
+
+assert(#data_files <= #colors, "only ".. #colors .." data files are supported")
+
+vname = vfile:match("([^/]+.v)$")
+
+print([[
+<html>
+<head>
+<title>]]..vname..[[</title>
+<style>]])
+for i,k in ipairs(colors) do
+  print(
+  ".time" .. i .. " {"..
+    "background-color: " .. k .. ";"..
+    "height: ".. 100 / #data_files .."%;"..
+	"top: " .. 100 / #data_files * (i - 1) .. "%;"..
+	"z-index: -1; position: absolute; opacity: 50%; }")
+end
+print([[.code {
+  z-index: 0;
+  position: relative;
+  border-style: solid;
+  border-color: transparent;
+  border-width: 1px;
+}
+.code:hover {
+  border-color: black;
+}
+pre {
+  display: inline;
+}
+</style>
+</head>
+<body>
+<h1>Timings for ]]..vname..[[</h1>
+<ol>
+]])
+for i,data_file in ipairs(data_files) do
+	print('<li style="background-color: '..colors[i]..'">' .. data_file .. "</li>")
+end
+print("</ol>")
+
+all_data = {}
+
+for _, data_file in ipairs(data_files) do
```

Comment:
If `ipairs ` is like Python's enumerate, why are you using it here just to throw away the number?

Reply:
it grants the order in increasing values of the key, `pairs` does not